### ISSUE TITLE
ngx-gauge Angular 9 / Ivy ready?

### ISSUE THREAD

**Issue by ankaufma** [ashish-chopra/ngx-gauge#65]
Today i tried to updated my project to angular 9/ivy rc.9. 
`ng update @angular/core @angular/cli --next` worked fine and i did get a success message.
`ng build` didn't work and it breaks with: 
```
ERROR in node_modules/ngx-gauge/gauge/gauge.d.ts:1:25 - error TS2724: Module '"C:/TMP/xxxxxxxx/node_modules/@angular/core/core"' has no exported member 'Renderer'. Did you mean 'Renderer2'?

1 import { SimpleChanges, Renderer, AfterViewInit, ElementRef, OnChanges, OnDestroy } from '@angular/core';
                          ~~~~~~~~

  node_modules/@angular/core/core.d.ts:5669:31
    5669 export declare abstract class Renderer2 {
                                       ~~~~~~~~~
    'Renderer2' is declared here.
```
There are much more other libraries involved but it points only to ngx-gauge.
As far as i understand it Angular/ngcc should substitute Renderer with Renderer2, but obviously it didn't. So is it a problem of the angular compiler or ngx-gauge? Has anybody tested ngx-gauge with angular 9?

**Comment by fernandocode** [comment]
Possibly this incompatibility would be resolved with [this PR](https://github.com/ashish-chopra/ngx-gauge/pull/59/files), but it would have to be analyzed and accepted. (@ashish-chopra)

**Comment by ankaufma** [comment]
@fernandocode you are right the PR solves this issue.
Meanwhile following powershell script does the job:

```
$s0 = "_renderer.setElementStyle"
$r0 = "_renderer.setStyle"

$s1 = "Renderer"
$r1 = "Renderer2"

ECHO "> Making ngx-gauge library ready for ivy compiler"
ECHO "> Replacing Renderer with Renderer2 ..."
Get-ChildItem "./node_modules/ngx-gauge/" -Recurse -Include *.ts, *.js, *.map, *.json, *.__ivy_ngcc_bak | ForEach-Object {
    (Get-Content $_.FullName) |
        ForEach-Object { $_ -replace [regex]::Escape($s0), $r0 } |
        Set-Content $_.FullName
}
Get-ChildItem "./node_modules/ngx-gauge/" -Recurse -Include *.ts, *.js, *.map, *.json, *.__ivy_ngcc_bak | ForEach-Object {
    (Get-Content $_.FullName) |
        ForEach-Object { $_ -replace [regex]::Escape($s1), $r1 } |
        Set-Content $_.FullName
}
```
i put it as `powershell.exe -command ./ngxGauge4IvyCompiler.ps1` in the postinstall script section of package.json and now it works

**Comment by ashish-chopra** [comment]
Thanks for the update @ankaufma and @fernandocode. I am going to activate this project again and spend some time on it in the coming few days and things will be fixed. @ankaufma, I am glad that you made a script that's a big lifesaver for people until I push commits in the main branch.

**Comment by ashish-chopra** [comment]
I have released Angular 9 version of ngx-gauge `v1.0.0` (no beta now). Kindly update your dependencies and try it and let me know if you face any errors.
